### PR TITLE
Update yq to v4

### PR DESCRIPTION
**Pull request by markusthoemmes**
Let's seeeee.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
/retest

Infra.

**Comment by markusthoemmes** [comment]
/retest

Take a guess why.

**Comment by mgencur** [comment]
Note to self. We'll need to do some updates in QE repositories when this is merged.

File: test/e2e-tests.sh
Comment (markusthoemmes) [inline test/e2e-tests.sh:3]: This will have to stay here for this to be mergable as the build-image will only update after a merge to main. We can then drop it in a subsequent PR.

**Comment by markusthoemmes** [comment]
/hold

Until after 1.16 cuts.

**Review by cardil — APPROVED** [review]
/lgtm

**Comment by matzew** [comment]
/lgtm

Others: feel free to unhold if you like

**Review by cardil — APPROVED** [review]
/lgtm

**Review by cardil — APPROVED** [review]
/lgtm

**Comment by matzew** [comment]
Should we wait untl we have cut the 1.17 ?

**Comment by markusthoemmes** [comment]
That's the plan, yes.

**Pull request closed without merging**
